dashboard/app/controllers/plc/enrollment_evaluations_controller.rb

```diff
@@ -1,25 +1,40 @@
 class Plc::EnrollmentEvaluationsController < ApplicationController
+  before_action :authorize_for_this_unit
+
   def perform_evaluation
-    authorize! :read, Plc::UserCourseEnrollment
     plc_unit_assignment = Plc::EnrollmentUnitAssignment.find(params[:unit_assignment_id])
 
-    if plc_unit_assignment.status != Plc::EnrollmentUnitAssignment::PENDING_EVALUATION
-      raise "Cannot perform evaluation - unit assignment is in state #{plc_unit_assignment.status}"
-    end
-
     @questions = plc_unit_assignment.plc_course_unit.plc_evaluation_questions
     @course_unit = plc_unit_assignment.plc_course_unit
   end
 
   def submit_evaluation
-    authorize! :read, Plc::UserCourseEnrollment
-    question_responses = params[:answerModuleList].split(',')
-    enrollment_unit_assignment = Plc::EnrollmentUnitAssignment.find(params[:unit_assignment_id])
+    # Eventually, the logic for more complex module assignments will go here. For now since we are naively assigning modules, just go to preview
+    redirect_to controller: :enrollment_evaluations, action: :preview_assignments, unit_assignment_id: params[:unit_assignment_id], enrolled_modules: params[:answer_module_list]
+  end
+
+  def preview_assignments
+    @enrollment_unit_assignment = Plc::EnrollmentUnitAssignment.find(params[:unit_assignment_id])
+    @enrolled_modules = Plc::LearningModule.where(id: params[:enrolled_modules].split(','))
+
+    if @enrolled_modules.where.not(plc_course_unit: @enrollment_unit_assignment.plc_course_unit).any?
+      redirect_to controller: :enrollment_evaluations, action: :perform_evaluation, unit_assignment_id: @enrollment_unit_assignment.id
+    end
+  end
 
-    modules_to_enroll_in = Plc::LearningModule.where(id: question_responses)
+  def confirm_assignments
+    modules_to_enroll_in = Plc::LearningModule.where(id: params[:learning_module_ids])
+    enrollment_unit_assignment = Plc::EnrollmentUnitAssignment.find(params[:unit_assignment_id])
```

Comment:
I have to get the enrollment unit assignment here - all I have is the enrollment unit assignment ID at this stage, not the object.

The modules to enroll in can get moved to model if I make the call take a list of module ids instead of modules. How large is the benefit?